app/kubemci/pkg/gcp/forwardingrule/forwardingrulesyncer_test.go

```diff
@@ -313,24 +309,69 @@ func TestListLoadBalancerStatuses(t *testing.T) {
 				},
 			},
 		},
+		{
+			"Should ignore the load balancer without status",
+			[]ensureFR{
+				{
+					lbName + "1",
+					ipAddr,
+					tpLink,
+					clusters,
+					false,
+					true,
+				},
+				{
+					lbName + "2",
+					ipAddr,
+					tpLink,
+					clusters,
+					false,
+					false,
+				},
+			},
+			map[string]status.LoadBalancerStatus{
+				lbName + "1": {
+					"",
+					lbName + "1",
+					clusters,
+					ipAddr,
+				},
+			},
+		},
 	}
-	namer := utilsnamer.NewNamer("mci1", lbName)
 	for i, c := range testCases {
 		glog.Infof("Test case: %d: %s", i, c.description)
 		frp := ingresslb.NewFakeLoadBalancers("" /*name*/, nil /*namer*/)
-		frs := NewForwardingRuleSyncer(namer, frp)
 		// Create all forwarding rules.
 		for _, fr := range c.forwardingRules {
+			namer := utilsnamer.NewNamer("mci1", fr.lbName)
+			frs := NewForwardingRuleSyncer(namer, frp)
 			if fr.https {
-				if err := frs.EnsureHttpsForwardingRule(fr.lbName, fr.ipAddr, fr.tpLink, fr.clusters, false /*force*/); err != nil {
+				if err := frs.EnsureHttpsForwardingRule(fr.lbName, fr.ipAddr, fr.tpLink, false /*force*/); err != nil {
 					t.Errorf("expected no error in ensuring https forwarding rule, actual: %v", err)
 				}
+				if fr.hasStatus {
+					typedFRS := frs.(*ForwardingRuleSyncer)
+					name := typedFRS.namer.HttpsForwardingRuleName()
+					if err := addStatus(typedFRS, fr.lbName, name, ipAddr, clusters); err != nil {
+						t.Fatalf("%s", err)
+					}
+				}
 			} else {
-				if err := frs.EnsureHttpForwardingRule(fr.lbName, fr.ipAddr, fr.tpLink, fr.clusters, false /*force*/); err != nil {
+				if err := frs.EnsureHttpForwardingRule(fr.lbName, fr.ipAddr, fr.tpLink, false /*force*/); err != nil {
 					t.Errorf("expected no error in ensuring http forwarding rule, actual: %v", err)
 				}
+				if fr.hasStatus {
+					typedFRS := frs.(*ForwardingRuleSyncer)
+					name := typedFRS.namer.HttpForwardingRuleName()
+					if err := addStatus(typedFRS, fr.lbName, name, ipAddr, clusters); err != nil {
+						t.Fatalf("%s", err)
```

Comment:
Are all these fatal errors really fatal? In other words, can't you run the next test case, i.e. the next iteration of the loop if this fails?